### ISSUE TITLE
Ambiguous batch format

### ISSUE THREAD

**Issue by Wollac** [ethereum-optimism/specs#24]
According to the [Channel Format](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/derivation.md#channel-format) section, `rlp_batches` is the concatenation of the RLP-encoded batches. This is somewhat misleading, as `rlp_batches` is not
```
batch1_version ++ rlp_encode([..]) ++ ... ++ batchn_version ++ rlp_encode([..])
```
(similar to a list of EIP-2718 transactions), but instead
```
rlp_encode(batch1_version ++ rlp_encode([..])) ++ ... ++ rlp_encode(batchn_version ++ rlp_encode([..]))
```
It should be made clear in the spec that `rlp_batches` is the concatenation of the RLP-encoded strings `batch_data`, where `batch_data = batch_version ++ content` as described in the section [Batch Format](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/derivation.md#batch-format).

And perhaps more importantly, it should be made clear what is supposed to happen during decoding, if the "byte envelope" is larger than `batch_version ++ content`. For example, if for `rlp_encode(batch1_version ++ rlp_encode([..]))` resulting in
```
b85000f84da0dbf6a80fef073de06add9b0d14026d6e5a86c85f6d102c36d3d8e9cf89c2afd3840109d8fea0438335a20d98863a4c0c97999eb2481921ccd28553eac6f913af7c12aec0410884647f5ea9c0
```
the following (adding a bunch of useless zeros) would also be valid:
```
b85800f84da0dbf6a80fef073de06add9b0d14026d6e5a86c85f6d102c36d3d8e9cf89c2afd3840109d8fea0438335a20d98863a4c0c97999eb2481921ccd28553eac6f913af7c12aec0410884647f5ea9c00000000000000000
```

If different implementations handle this decoding differently, it could lead to a chain fork.

**Comment by zhiqiangxu** [comment]
what do you mean by `batchn_version`, IIRC the byte version is only attached to the compressed channel frames, like [here](https://github.com/ethereum-optimism/optimism/blob/d2263b19958f7f1e738ff79dae7ebb240cca1375/op-batcher/batcher/tx_data.go#L48).

**Comment by zhiqiangxu** [comment]
For singular batch, it's `rlp_encode(SingularBatchType ++ rlp_encode(SingularBatch)) ++ ... ++ rlp_encode(SingularBatchType ++ rlp_encode(SingularBatch))`.

For span batch, it's `rlp_encode(SpanBatchType ++ RawSpanBatch.encode(RawSpanBatch))`.

**Comment by Wollac** [comment]
Exactly. This is how it should be described in the spec to be 100% clear and unambiguous. It should also be described what happens if the buffer contains additional data after the outer `rlp_encode(...)`. This is probably an error?

**Comment by zhiqiangxu** [comment]
This is what exactly [this pr](https://github.com/ethereum-optimism/optimism/pull/10290) is trying to do, but it may cause chain split.
